src/DotNet.Testcontainers/Clients/DockerNetworkOperations.cs

```diff
@@ -0,0 +1,88 @@
+namespace DotNet.Testcontainers.Clients
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Linq;
+  using System.Threading;
+  using System.Threading.Tasks;
+  using Docker.DotNet.Models;
+  using DotNet.Testcontainers.Networks.Configurations;
+  using DotNet.Testcontainers.Services;
+  using Microsoft.Extensions.Logging;
+
+  internal sealed class DockerNetworkOperations : DockerApiClient, IDockerNetworkOperations
+  {
+    private static readonly ILogger<DockerNetworkOperations> Logger = TestcontainersHostService.GetLogger<DockerNetworkOperations>();
+
+    public DockerNetworkOperations(Uri endpoint) : base(endpoint)
+    {
+    }
+
+    public async Task<IEnumerable<NetworkResponse>> GetAllAsync(CancellationToken ct = default)
+    {
+      return (await this.Docker.Networks.ListNetworksAsync(new NetworksListParameters(), ct)
+        .ConfigureAwait(false)).ToArray();
+    }
+
+    public async Task<IEnumerable<NetworkResponse>> GetOrphanedObjects(CancellationToken ct = default)
+    {
+      var filters = new FilterByProperty {{"label", $"{TestcontainersClient.TestcontainersCleanUpLabel}=true"}, {"dangling", "true"}};
```

Comment:
Just thinking about cleaning up resources from previous runs, we may need to think of something more holistic than this. Since the network is created before the containers are started, the containers will likely still be connected to the network so we they won't be dangling.

Because of this, we'll need to clean up the old containers before we clean up the old networks. Not sure how to approach this one. I'll think about it some more.

Reply:
Yeah, I haven't thought about that.... We probably can't implement it that way at all. A network resource has no status. We do not know at all when it's no longer needed.

Reply:
I removed the part for now. I don't know how to remove orphaned networks properly. This is tricky. I think it requires an overall solution incl. containers.